rest_server/table_config.py

```diff
@@ -69,248 +56,292 @@ class _ColumnConfigTypedDict(TypedDict, total=False):
+                "border_left": True,
+                "on_the_fly": True,
+                "tooltip": "TOTAL-ROWS ONLY: FTE totals to the right refer to this category.",
+            },
+            NSF_MO_CORE: {
+                "width": 50,
+                "funding_source": True,
+                "non_editable": True,
+                "hidden": True,
+                "numeric": True,
+                "on_the_fly": True,
+                "tooltip": tooltip_funding_source_value,
+            },
+            NSF_BASE_GRANTS: {
+                "width": 50,
+                "funding_source": True,
+                "non_editable": True,
+                "hidden": True,
+                "numeric": True,
+                "on_the_fly": True,
+                "tooltip": tooltip_funding_source_value,
+            },
+            US_IN_KIND: {
+                "width": 50,
+                "funding_source": True,
+                "non_editable": True,
+                "hidden": True,
+                "numeric": True,
+                "on_the_fly": True,
+                "tooltip": tooltip_funding_source_value,
+            },
+            NON_US_IN_KIND: {
+                "width": 50,
+                "funding_source": True,
+                "non_editable": True,
+                "hidden": True,
+                "numeric": True,
+                "on_the_fly": True,
+                "tooltip": tooltip_funding_source_value,
+            },
+            GRAND_TOTAL: {
+                "width": 50,
+                "numeric": True,
+                "non_editable": True,
+                "hidden": True,
+                "border_left": True,
+                "on_the_fly": True,
+                "tooltip": "This is is the total of the four FTEs to the left.",
+            },
+            ID: {"width": 0, "non_editable": True, "border_left": True, "hidden": True},
+            TIMESTAMP: {
+                "width": 100,
+                "non_editable": True,
+                "border_left": True,
+                "hidden": True,
+                "tooltip": f"{TIMESTAMP} (you may need to refresh to reflect a recent update)",
+            },
+            EDITOR: {
+                "width": 100,
+                "non_editable": True,
+                "hidden": True,
+                "tooltip": f"{EDITOR} (you may need to refresh to reflect a recent update)",
+            },
+        }
+
+    @property
+    def icecube_institutions(self) -> Dict[str, InstitutionMeta]:
+        """Grab the master list of institutions along with their details.
+
+        NOTE: `requests.get()` + `eval()` is a stopgap measure until
+        the Keycloak REST Service is operational.
+        """
+        file_url = "https://raw.githubusercontent.com/WIPACrepo/keycloak-rest-services/master/keycloak_setup/institution_list.py"
+        resp = requests.get(file_url)
+
+        glo: Dict[str, Any] = {}
+        eval(compile(resp.text, "<string>", "exec"), {}, glo)  # pylint:disable=W0123
```

Comment:
This is pretty evil.  Really not sure this is a good idea to do all the time.  Maybe once in a build process?  You could download the python file at build time, then just import it.

Reply:
I agree it's not good.

I couldn't think of any other way to dynamically update the list without a rebuild. _Requiring a reload on rebuild isn't terrible, however._ In production currently, the dependency packages are getting cached, so this wasn't an option when the list was imported from krs.